$NetBSD: patch-Makefile,v 1.1 2014/03/28 03:48:22 phonohawk Exp $

* Don't install compressed man pages.

* Pass LDFLAGS to the linker, or it may find a wrong libcrypto.

--- Makefile.orig	2014-03-27 03:34:47.000000000 +0000
+++ Makefile
@@ -68,25 +68,23 @@ endif
 APPS=edge
 APPS+=supernode
 
-DOCS=edge.8.gz supernode.1.gz n2n_v2.7.gz
-
-all: $(APPS) $(DOCS)
+all: $(APPS)
 
 edge: edge.c $(N2N_LIB) n2n_wire.h n2n.h Makefile
-	$(CC) $(CFLAGS) edge.c $(N2N_LIB) $(LIBS_EDGE) -o edge
+	$(CC) $(CFLAGS) $(LDFLAGS) edge.c $(N2N_LIB) $(LIBS_EDGE) -o edge
 
 test: test.c $(N2N_LIB) n2n_wire.h n2n.h Makefile
-	$(CC) $(CFLAGS) test.c $(N2N_LIB) $(LIBS_EDGE) -o test
+	$(CC) $(CFLAGS) $(LDFLAGS) test.c $(N2N_LIB) $(LIBS_EDGE) -o test
 
 supernode: sn.c $(N2N_LIB) n2n.h Makefile
-	$(CC) $(CFLAGS) sn.c $(N2N_LIB) $(LIBS_SN) -o supernode
+	$(CC) $(CFLAGS) $(LDFLAGS) sn.c $(N2N_LIB) $(LIBS_SN) -o supernode
 
 benchmark: benchmark.c $(N2N_LIB) n2n_wire.h n2n.h Makefile
-	$(CC) $(CFLAGS) benchmark.c $(N2N_LIB) $(LIBS_SN) -o benchmark
+	$(CC) $(CFLAGS) $(LDFLAGS) benchmark.c $(N2N_LIB) $(LIBS_SN) -o benchmark
 
 ifeq ($(SNM), yes)
 test_snm: sn_multiple_test.c $(N2N_LIB) n2n.h Makefile
-	$(CC) $(CFLAGS) sn_multiple_test.c $(N2N_LIB) $(LIBS_SN) -o test_snm
+	$(CC) $(CFLAGS) $(LDFLAGS) sn_multiple_test.c $(N2N_LIB) $(LIBS_SN) -o test_snm
 endif
 
 .c.o: n2n.h n2n_keyfile.h n2n_transforms.h n2n_wire.h twofish.h Makefile
@@ -103,13 +101,13 @@ version.o: Makefile
 	$(CC) $(CFLAGS) -DN2N_VERSION='"$(N2N_VERSION)"' -DN2N_OSNAME='"$(N2N_OSNAME)"' -c version.c
 
 clean:
-	rm -rf $(N2N_OBJS) $(N2N_LIB) $(APPS) $(DOCS) test *.dSYM *~
+	rm -rf $(N2N_OBJS) $(N2N_LIB) $(APPS) test *.dSYM *~
 
-install: edge supernode edge.8.gz supernode.1.gz n2n_v2.7.gz
+install: edge supernode
 	echo "MANDIR=$(MANDIR)"
 	$(MKDIR) $(SBINDIR) $(MAN1DIR) $(MAN7DIR) $(MAN8DIR)
 	$(INSTALL_PROG) supernode $(SBINDIR)/
 	$(INSTALL_PROG) edge $(SBINDIR)/
-	$(INSTALL_DOC) edge.8.gz $(MAN8DIR)/
-	$(INSTALL_DOC) supernode.1.gz $(MAN1DIR)/
-	$(INSTALL_DOC) n2n_v2.7.gz $(MAN7DIR)/
+	$(INSTALL_DOC) edge.8 $(MAN8DIR)/
+	$(INSTALL_DOC) supernode.1 $(MAN1DIR)/
+	$(INSTALL_DOC) n2n_v2.7 $(MAN7DIR)/
